scrapper.py

```diff
@@ -25,39 +36,159 @@ class Crawler:
+            if refs_maker in text_from_pdf:
+                text_pdf_list = text_from_pdf.split(refs_maker)
+                del text_pdf_list[-1]
+                text_from_pdf = ' '.join(text_pdf_list)
+        self.article.text = text_from_pdf
+
+    def _fill_article_with_meta_information(self, article_bs):
+        issue_number = []
+        month_dict = {2: '03', 3: '06', 0: '09', 1: '12'}
+        try:
+            author = article_bs.find('p', {"class": "articles-item-author"}).find('b')
+            self.article.author = author.text
+        except AttributeError:
+            self.article.author = 'No data'
+        title = article_bs.find('h2', {"class": "articles-item-name"})
+        self.article.title = title.text
+        topics = article_bs.find('div', {"class": "articles"}).find_all('p')
+        try:
+            self.article.topics = topics[2].text
+        except IndexError:
+            self.article.topics = 'No data'
+        year = article_bs.find('meta', {'name': 'citation_publication_date'})['content']
+        issue_title = article_bs.find('meta', {'name': 'citation_journal_title'})['content']
+        for letter in issue_title:
+            if letter.isdigit():
+                issue_number.append(letter)
+        month = month_dict[int(''.join(issue_number)) % 4]
+        text_date = month + '.' + year
+        date = datetime.datetime.strptime(text_date, '%m.%Y')
+        self.article.date = date
 
 
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    base_path_p = pathlib.Path(base_path)
+    if os.path.exists(base_path_p):
+        shutil.rmtree(base_path_p)
+        base_path_p.mkdir(parents=True)
+    else:
+        base_path_p.mkdir(parents=True)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path) as path_thing_idk:
+        config_dict = json.load(path_thing_idk)
+    seed_urls = config_dict['seed_urls']
+    max_articles = config_dict["total_articles_to_find_and_parse"]
+    if not isinstance(max_articles, int) or max_articles <= 0:
+        raise IncorrectNumberOfArticlesError
+    if max_articles > 150:
+        raise NumberOfArticlesOutOfRangeError
+    if not isinstance(seed_urls, list):
+        raise IncorrectURLError
+    if len(seed_urls) <= 1:
+        raise IncorrectURLError
+    for seed_url in seed_urls:
+        if not isinstance(seed_url, str):
+            raise IncorrectURLError
+        if not seed_url.startswith('https://'):
+            raise IncorrectURLError
+
+    return seed_urls, max_articles
 
 
 if __name__ == '__main__':
     # YOUR CODE HERE
-    pass
+    i = 0
+    s_urls, max_as = validate_config(CRAWLER_CONFIG_PATH)
+    prepare_environment(ASSETS_PATH)
+    crawler = Crawler(s_urls, max_as)
+    crawler.find_articles()
+    print(len(crawler.urls))
+    for art_url in crawler.urls:
```

Comment:
there should be a more elegant solution for getting access to the index of current element in the loop